City-of-Helsinki/infofinland-ui
# Ifu 419 442/sitemap and helsinkilogo

- Error 500 -page
-  Skip-to-content-link
- Sitemap scaffold (with demo content for now)
- Search UI improvements: font sizes, button alignments etc

**How to test Error500**
- Checkout branch, run `yarn && yarn dev`
- Go to localhost:3000/500 to see the error page.

OR, throw an error from some file, for example: 
- Open article-sitemap.xml.jsx, 
- throw error before returning sitemap. 
- run `yarn build && yarn start`
- go to article-sitemap.xml. Should give Error 500 page.


**How to test sitemap**
Test that the following files are served and that static sitemap is generated in build-time.
For now the static sitemap uses ENV variable to define it's own url at build time. This might need some tweaking depending on Azure settings.,

- run `SITE_URL=localhost:3000 yarn build & yarn start`. 
- go to `http://localhost:3000/sitemap.xml` for static files. See that it does not include system files like Lang404 (config in next-sitemap.js)
- go to  `http://localhost:3000/article-sitemap.xml` ,fort dynamic routes. This is just static code atm. Must be connected to drupal later
-robots.txt , for robot stuff. Must contain both sitemaps.



**How to test Skip to content**

- Open site in browser
- use tab-key to navigate. Click somewhere near logo, hit "Tab", focus ring should appear on logo. 
- Hit Tab again. A "Skip to content" link should appear, from behind the logo to the right of logo on desktop, below logo on mobile.
